libs/portals/admin/regulations-admin/src/state/validations.ts

```diff
@@ -0,0 +1,313 @@
+
+          if (newMentions.includes(impact.name)) {
+            delete impact.error
+          } else {
+            impact.error = errorMsgs.impactingUnMentioned
+          }
+        })
+      })
+    }
+  }
+}
+
+// ---------------------------------------------------------------------------
+
+/**
+ * Collection of "guessing" logic — setting/updating values/fields that
+ * are 100% dependant on other values.
+ */
+export const derivedUpdates: {
+  [Prop in RegDraftFormSimpleProps]?: (
+    state: DraftingState,
+    newValue: RegDraftForm[Prop]['value'],
+  ) => RegDraftForm[Prop]['value'] | null | void
+} = {
+  /**
+   * Derive regulation `type` from wording of title.
+   * Re-validate impacts (search for "mentions" of regulation names)
+   */
+  title: (state: DraftingState, newTitle) => {
+    const { type, text } = state.draft
+
+    /**
+     * Type is no longer derived. It is set on creation.
+     * */
+    // const derivedType = findRegulationType(newTitle)
+    // updateFieldValue(type, derivedType)
+
+    if (!newTitle) {
+      type.showError = undefined
+    }
+    updateImpacts(state.draft, newTitle, text.value)
+  },
+
+  /**
+   * Re-validate impacts (search for "mentions" of regulation names)
+   */
+  text: (state, newValue) => {
+    const { title } = state.draft
+    updateImpacts(state.draft, title.value, newValue)
+  },
+
+  /**
+   * Parse the signature block and read the ministry name and signature date.
+   *
+   * This should effectively enforce conventional/standardized wording
+   * of beginning of the signature block.
+   */
+  signatureText: (state, newValue) => {
+    const draft = state.draft
+    const { ministryName, signatureDate } = findSignatureInText(newValue)
+
+    updateFieldValue(
+      draft.signatureDate,
+      signatureDate && new Date(signatureDate),
+    )
+
+    const ministryNameNormalized = ministryName?.toLowerCase().replace(/-/g, '')
+    const knownMinistry = ministryNameNormalized
+      ? state.ministries.find(
+          (m) =>
+            m.name.toLowerCase().replace(/-/g, '') === ministryNameNormalized,
+        )
+      : undefined
+    // prefer official name over typed name (ignores casing, and punctuation differernces)
+    const newMinistryName = knownMinistry ? knownMinistry.name : ministryName
+    updateFieldValue(draft.ministry, newMinistryName)
+
+    // TODO: Also match the derived ministry up with original draft author's
+    // ministry connnection (i.e. their "place of work")
+    // and issue a WARNING if the two ministry values don't match up.
+    const hasMinistryError = ministryName && !knownMinistry
+    const ministryError = hasMinistryError
+      ? errorMsgs.ministryUnknown
+      : undefined
+    draft.ministry.error = ministryError || draft.ministry.error
+    draft.ministry.showError = hasMinistryError || draft.ministry.showError
+  },
+
+  idealPublishDate: (state, newValue) => {
+    // FIXME: TODO: lower-boundry-check explicitly set impact dates.
+  },
```

Comment:
Fixme-Todo!

Reply:
This wil be part of publishing, which is the next chapter.